perf/counter.go

```diff
@@ -229,18 +251,69 @@ func (c Count) Value() (float64, string) {
 // only have a single Event, this is faster and more ergonomic than
 // [Counter.ReadGroup].
 func (c *Counter) ReadOne() (Count, error) {
-	// TODO: Use RDPMC when possible.
 	if c == nil {
 		return Count{}, nil
 	}
 
+	// Use RDPMC when possible.
+	cnt, ok := readPMC(c.mmap[0])
+	if ok {
+		return cnt, nil
+	}
+
 	var cs [1]Count
 	if err := c.ReadGroup(cs[:]); err != nil {
 		return Count{}, err
 	}
 	return cs[0], nil
 }
 
+// Returns false if reading from PMC is not currently available. Note that this
+// may change from one call to the next due to e.g., multiplexing.
+func readPMC(mmap *unix.PerfEventMmapPage) (c Count, ok bool) {
+	if runtime.GOARCH != "amd64" {
+		return c, false
+	}
+
+	if mmap.Compat_version > perfEventMmapPageVersion {
+		return c, false
+	}
+
+	// See Linux tools/lib/perf/mmap.c:perf_mmap__read_self for reference.
```

Comment:
Based on perf_mmap__read_self, it looks like we need to check `cap_user_time` and `cap_user_time_short` and adjust the enabled and running time accordingly.